### PR TITLE
Port search page and search bar from search-assets

### PR DESCRIPTION
**Pull request by Bouncey**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.

### CODE REVIEW

**Comment by RandellDawson** [comment]
@Bouncey If I want to test this PR, is there a way to update the indexes, so I can see how the search functionality would work with all the changes and additions to the repo's guide and challenge files?

**Comment by Bouncey** [comment]
It should work with the current indexes.

Re-indexing is a task unto itself, and something I wish to tackle separlately.

**Comment by RandellDawson** [comment]
Oh OK.  There are about 5-10 open issues related to the indexes not being updated in a while, so I guess those will be resolved once you perform that separate task.  I was just anxious to maybe be able to close those too.

File: client/src/components/search/searchPage/SearchPageHits.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  return currentRefinement && !isHitsEmpty ? (
+    <div className='ais-Hits search-page'>
+      <ul className='ais-Hits-list'>
+        {nonQuerySuggestionHits.map(({ hits: results, index }) =>
+          results.map(result => (
+            <a
+              href={buildUrl(index, result)}
+              key={result.objectID}
+              target='_blank'
```
Comment (raisedadead) [inline client/src/components/search/searchPage/SearchPageHits.js]:
```suggestion
              rel='noopener noreferrer'
              target='_blank'
```

Resolve the warning.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Review by raisedadead — CHANGES_REQUESTED** [review]
There are few styling suggestions as well that I wanted to make based on https://github.com/freeCodeCamp/freeCodeCamp/issues/35103

Do you want to tackle them here, or wait for a future PR.

File: client/package-lock.json [hunk 2]
```diff
@@ -2492,7 +2492,7 @@
     },
     "bl": {
       "version": "1.2.2",
-      "resolved": "https://registry.npmjs.org/bl/-/bl-1.2.2.tgz",
+      "resolved": "http://registry.npmjs.org/bl/-/bl-1.2.2.tgz",
```
Comment (raisedadead) [inline client/package-lock.json]: Something weird is happening with the lockfile. You may want to run `npm install --prefer-online` on root.

File: client/src/components/Header/index.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Media from 'react-media';
-import FCCSearch from 'react-freecodecamp-search';
 
+import FCCSearch from '../search/searchBar/SearchBar';
```
Comment (raisedadead) [inline client/src/components/Header/index.js]: Is there a reason we want to call it `FCCSearch` vs `Search`

**Comment by ahmadabdolsaheb** [comment]
@raisedadead I can sort the styles out as soon as this is merged.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Thanks @Bouncey for the intial work and @ValeraS for taking this up. I have merged this to keep going forth.

**Pull request merged**